lib/matplotlib/backends/backend_pdf.py

```diff
@@ -2115,116 +2104,77 @@ def draw_text(self, gc, x, y, s, prop, angle, ismath=False, mtext=None):
                              -math.sin(a), math.cos(a),
                              x, y, Op.concat_matrix)
-
-            # Output all the 1-byte characters in a BT/ET group.
+            # Emit all the 1-byte characters in a BT/ET group.
             self.file.output(Op.begin_text,
-                             self.file.fontName(prop),
-                             fontsize,
-                             Op.selectfont)
-            newx = oldx = 0
-            for chunk_type, chunk in chunks:
-                if chunk_type == 1:
-                    self._setup_textpos(newx, 0, 0, oldx, 0, 0)
-                    self.file.output(self.encode_string(chunk, fonttype),
-                                     Op.show)
-                    oldx = newx
-                # Update newx to include the advance from this chunk,
-                # regardless of its mode...
-                for char_idx, char_x in _text_layout.layout(
-                        chunk, font, x0=newx, kern_mode=KERNING_UNFITTED):
-                    pass
-                newx = char_x + (  # ... including the last character's advance
-                    font.load_glyph(char_idx, flags=LOAD_NO_HINTING)
-                    .linearHoriAdvance / 65536)
+                             self.file.fontName(prop), fontsize, Op.selectfont)
+            prev_start_x = 0
+            for start_x, chars in singlebyte_chunks:
+                self._setup_textpos(start_x, 0, 0, prev_start_x, 0, 0)
+                self.file.output(self.encode_string(''.join(chars), fonttype),
+                                 Op.show)
+                prev_start_x = start_x
             self.file.output(Op.end_text)
-
-            # Then output all the 2-byte characters.
-            newx = 0
-            for chunk_type, chunk in chunks:
-                for char_idx, char_x in _text_layout.layout(
-                        chunk, font, x0=newx, kern_mode=KERNING_UNFITTED):
-                    if chunk_type == 2:
-                        glyph_name = font.get_glyph_name(char_idx)
-                        self.file.output(Op.gsave)
-                        self.file.output(0.001 * fontsize, 0,
-                                         0, 0.001 * fontsize,
-                                         char_x, 0, Op.concat_matrix)
-                        name = self.file._get_xobject_symbol_name(
-                            font.fname, glyph_name)
-                        self.file.output(Name(name), Op.use_xobject)
-                        self.file.output(Op.grestore)
-                newx = char_x + (  # ... including the last character's advance
-                    font.load_glyph(char_idx, flags=LOAD_NO_HINTING)
-                    .linearHoriAdvance / 65536)
-
+            # Then emit all the multibyte characters, one at a time.
+            for start_x, (char,) in multibyte_chunks:
+                glyph_name = font.get_glyph_name(
+                    font.get_char_index(ord(char)))
```

Comment:
You can't just put `char_idx` into `multibyte_chunks` instead?

Reply:
same as https://github.com/matplotlib/matplotlib/pull/14886/files#r307636753, now changed